### PR TITLE
feat: Split orders by referrer

### PR DESCRIPTION
**Pull request by 0xBunta**

### CODE REVIEW

File: src/LooksRareAggregator.ts [hunk 1]
```diff
@@ -133,8 +133,24 @@ export class LooksRareAggregator {
     return transformSeaportListings(this.chainId, listings, proxyAddress);
   }
 
-  private async transformLooksRareV2Listings(listings: MakerOrderFromAPI[]): Promise<TradeData> {
-    return await transformLooksRareV2Listings(this.chainId, this.signer, listings, this.addresses.LOOKSRARE_V2_PROXY);
+  private async transformLooksRareV2Listings(listings: MakerOrderFromAPI[]): Promise<Array<TradeData>> {
+    //Split orders into different TradeData(Proxy call) if orders have different referrer addresses
```
Comment (0xhiroshi) [inline src/LooksRareAggregator.ts:137]:
```suggestion
    // Split orders into different TradeData (Proxy call) if orders have different referrer addresses
```

File: src/__tests__/aggregatorExecuteLooksRareV2.test.ts [hunk 2]
```diff
@@ -45,8 +45,8 @@ describe("LooksRareAggregator class", () => {
     const makerOrder: Maker = {
       quoteType: QuoteType.Ask,
       globalNonce: 0,
-      subsetNonce: 0,
-      orderNonce: 0,
+      subsetNonce: nonce || 0,
+      orderNonce: nonce || 0,
```
Comment (0xhiroshi) [inline src/__tests__/aggregatorExecuteLooksRareV2.test.ts:49]: Can you remind me whether you can set a default value for `nonce` in the function argument definition?

File: src/__tests__/aggregatorExecuteLooksRareV2.test.ts [hunk 3]
```diff
@@ -66,21 +66,37 @@ describe("LooksRareAggregator class", () => {
...
+    const referrerFromAPI: Referrer = {
+      rate: 0,
+      address: referrer?.address || constants.HashZero,
+    };
     // Fake an order from the API
-    const makerOrderFromAPI: MakerOrderFromAPI = {
+    return {
       merkleTree,
       signature,
+      referrer: referrer ? referrerFromAPI : undefined,
```
Comment (0xhiroshi) [inline src/__tests__/aggregatorExecuteLooksRareV2.test.ts:77]: so if referrer address is blank then there is no point defining `referrerFromAPI` even?

File: src/__tests__/aggregatorExecuteLooksRareV2.test.ts [hunk 4]
```diff
@@ -66,21 +66,37 @@ describe("LooksRareAggregator class", () => {
...
       ...makerOrder,
     };
+  };
+
+  const executeAggregator = async (
+    maker: SignerWithAddress,
+    collection: Contract,
+    makerOrderFromAPIs: Array<MakerOrderFromAPI>
```
Comment (0xhiroshi) [inline src/__tests__/aggregatorExecuteLooksRareV2.test.ts:85]:
```suggestion
    makerOrdersFromAPI: Array<MakerOrderFromAPI>
```

File: src/__tests__/aggregatorExecuteLooksRareV2.test.ts (hunk 2 above)
Comment (0xBunta) [reply src/__tests__/aggregatorExecuteLooksRareV2.test.ts:49]: Yes, it works and I fixed it!

File: src/__tests__/aggregatorExecuteLooksRareV2.test.ts (hunk 3 above)
Comment (0xBunta) [reply src/__tests__/aggregatorExecuteLooksRareV2.test.ts]: True, fixed!

**Pull request merged**